internal/commands/server/k8s_e2e_test.go

```diff
@@ -880,6 +873,41 @@ func conditionInSync(conditions []meta.Condition) bool {
 	return false
 }
 
+func createGateway(ctx context.Context, t *testing.T, cfg *envconf.Config, gatewayName string, gc *gateway.GatewayClass) *gateway.Gateway {
+	t.Helper()
+
+	namespace := e2e.Namespace(ctx)
+	gatewayNamespace := gateway.Namespace(namespace)
+
+	resources := cfg.Client().Resources(namespace)
+
+	gw := &gateway.Gateway{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      gatewayName,
+			Namespace: namespace,
+		},
+		Spec: gateway.GatewaySpec{
+			GatewayClassName: gateway.ObjectName(gc.Name),
+			Listeners: []gateway.Listener{{
+				Name:     "https",
+				Port:     gateway.PortNumber(443),
```

Comment:
@nathancoleman So this is why your tests are failing. When you extracted this helper you set all of the listeners to bind to port `443` -- the problem is that to actually route through the listener that's deployed in `kind` from the localhost you need to do all sorts of port-forwarding in the `kind` cluster itself. If you want to keep this extracted helper, then you'll need to parameterize it to take a port and then pass in `e2e.HTTPPort(ctx)` to the Gateway that you actually want to check the routing through, which I believe is currently on line 302.

Reply:
Good catch!